arbos/l1pricing/l1pricing.go

```diff
@@ -249,22 +250,22 @@ func (ps *L1PricingState) UpdateForSequencerSpending(stateDb vm.StateDB, updateT
 	}
 
 	// allocate funds to this update
-	collectedSinceUpdate := stateDb.GetBalance(L1PricerFundsPoolAddress)
-	fundsToMove := new(big.Int).Div(new(big.Int).Mul(collectedSinceUpdate, allocFractionNum), allocFractionDenom)
-	stateDb.SubBalance(L1PricerFundsPoolAddress, fundsToMove)
-	availableFunds = new(big.Int).Add(availableFunds, fundsToMove)
+	collectedSinceUpdate := statedb.GetBalance(L1PricerFundsPoolAddress)
+	fundsToMove := am.BigDivByUint(am.BigMulByUint(collectedSinceUpdate, updateTimeDelta), timeDelta)
+	statedb.SubBalance(L1PricerFundsPoolAddress, fundsToMove)
+	availableFunds = am.BigAdd(availableFunds, fundsToMove)
 
 	// update amounts due
 	perUnitReward, err := ps.PerUnitReward()
 	if err != nil {
 		return err
 	}
-	fundsDueToSequencer = new(big.Int).Add(fundsDueToSequencer, weiSpent)
+	fundsDueToSequencer = am.BigAdd(fundsDueToSequencer, weiSpent)
 	if err := ps.SetFundsDueToSequencer(fundsDueToSequencer); err != nil {
 		return err
 	}
-	newRewards := new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(perUnitReward)), allocFractionNum), allocFractionDenom)
-	fundsDueForRewards = new(big.Int).Add(fundsDueForRewards, newRewards)
+	newRewards := am.SaturatingUMul(updateTimeDelta, perUnitReward) / timeDelta
```

Comment:
The SaturatingUMul might possibly saturate the first time this is invoked, if the time advances from zero to the real time.  After that it shouldn't be a problem.  It might be OK to allow it to saturate once, in order to go faster in the common case.